packages/frontend/src/bootstrap.tsx

```diff
@@ -45,14 +49,21 @@ export const Root = ({
     return (
         <HashRouter>
             <ThemeUpdater />
-            <CarrotCoreProvider
-                supportedChains={supportedChains || standaloneSupportedChains}
-                providers={providers || standaloneProviders}
-                getConnectors={connectors || getStandaloneConnectors}
-                ipfsGatewayURL={ipfsGatewayURL}
-            >
-                <App customBaseURL={customBaseURL} templateId={templateId} />
-            </CarrotCoreProvider>
+            <QueryClientProvider client={queryClient}>
```

Comment:
Pretty sure the plan is to eventually also use React Query in the `@carrot-kpi/react` lib. That's why I think it's better to have the Query Provider there, inside the `CarrotCoreProvider`.

Reply:
Yes, but then we have to move all the hooks there, otherwise it will not be possible to use it in places like `frontend/src/hooks/useLatestKPITokens.ts`.

Reply:
>Yes, but then we have to move all the hooks there, otherwise it will not be possible to use it in places like frontend/src/hooks/useLatestKPITokens.ts.

Wait that's not true. The `CarrotCoreProvider` wraps the frontend's `App` so we can use React query in both the React app and the frontend. That's the same principle we use for the wagmi client, which is shared between the React lib and the frontend. In order to do that it's important to add React Query to peer dependencies in the React lib and to the dependencies in the frontend.

Reply:
Yeah, it makes sense it works on both, but I couldn't make it work. I'll try again.

Reply:
wasn't able to do this, I tried to add peer dependencies as it is with wagmi but it fails.

`React package`
<img width="563" alt="image" src="https://user-images.githubusercontent.com/5664434/220243903-4fe83d08-c146-462f-9759-b7df69b155ab.png">
**frontend package**
<img width="592" alt="image" src="https://user-images.githubusercontent.com/5664434/220243927-0efb0537-4537-4c05-8cb4-e16afadaaee9.png">
 
also added `QueryClientProvider` to react package and remove it from bootstrap.

<img width="648" alt="image" src="https://user-images.githubusercontent.com/5664434/220244785-87c7d057-f942-4310-b56b-1047e68436d0.png">


fails with:

`No QueryClient set, use QueryClientProvider to set on
`